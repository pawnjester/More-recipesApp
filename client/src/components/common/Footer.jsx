import React from 'react';

const Footer = () => (
  <div>
    <footer id="main-footer" className="bg-danger navbar-static-bottom ">
      <div className="container">
        <div className="row">
          <div className="col-md-6 col-sm-12 text-center">
            <div className="py-2 text-white">
              <h1 className="h3">More-Recipes</h1>

            </div>
          </div>
          <div className="col-md-6 col-sm-12 text-center">
            <div className="py-2 text-white">
              <h1 className="h6">Andela Bootcamp cycle XXVII</h1>
              <p>Copyright &copy; 2017</p>
            </div>
          </div>

        </div>

      </div>

    </footer>
  </div>
);

export default Footer;

