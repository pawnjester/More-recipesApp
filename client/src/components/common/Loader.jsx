import React from 'react';
import { RingLoader } from 'react-spinners';


export const Loader = props => (
  <div className="sweet-loading loader-center">
    <RingLoader
      color="#e50000"
      loading={props.Loading}
    />
  </div>
);

export default Loader;
