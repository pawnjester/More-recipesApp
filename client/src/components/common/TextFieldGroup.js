import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';


const TextFieldGroup = ({field,value, error, type, onChange, placeholder}) => {
  return (
    <div className= {classnames("form-group", {'has-error': error})}>
                      <input
                      value = {value} 
                      onChange = {onChange}                     
                       type={type} 
                      name = {field}
                      placeholder = {placeholder}
                       className="form-control form-control-lg" 
                       /> 
                       
                       {error && <span className ="help-block"> {error}</span>}
    </div>
  );
}

TextFieldGroup.propTypes = {
  field: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  error: PropTypes.string,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired


}

TextFieldGroup.defaultProps = {
  type: 'text'
}


export default TextFieldGroup;