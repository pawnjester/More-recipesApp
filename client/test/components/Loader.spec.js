import React from 'React';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import sinon from 'sinon';
import { provider } from 'react-redux';
import { Loader } from '../../src/components/common/Loader';
import store from '../../src/store/store';

configure({ adapter: new Adapter() });

function setup() {
  const shallowWrapper = shallow(<Loader /> );
  return {
    shallowWrapper,
  }
}

describe('Test for Loader', () => {
  it('should render correctly', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
  });
});
