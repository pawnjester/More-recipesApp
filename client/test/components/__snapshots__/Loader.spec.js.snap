// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Loader should render correctly 1`] = `
<div
  className="sweet-loading loader-center"
>
  <onlyUpdateForKeys(Loader)
    color="#e50000"
    loading={true}
    size={60}
  />
</div>
`;
